src/exec.h

```diff
@@ -0,0 +1,45 @@
+#ifndef AKTUALIZR_LITE_EXEC_H_
+#define AKTUALIZR_LITE_EXEC_H_
+
+#include <boost/asio/io_service.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/format.hpp>
+#include <boost/process.hpp>
+
+template <typename... Args>
+static void exec(const std::string& cmd, const std::string& err_msg_prefix, Args&&... args) {
+  // Implementation is based on test_utils.cc:Process::spawn that has been proven over time
+  std::future<std::string> err_output;
+  std::future<int> child_process_exit_code;
+  boost::asio::io_service io_service;
+
+  try {
+    boost::process::child child_process(cmd, boost::process::std_err > err_output,
+                                        boost::process::on_exit = child_process_exit_code, io_service,
+                                        std::forward<Args>(args)...);
+
+    io_service.run();
+
+    bool wait_successfull = child_process.wait_for(std::chrono::seconds(900));
+    if (!wait_successfull) {
+      throw std::runtime_error("Timeout occured while waiting for a child process completion");
+    }
+
+  } catch (const std::exception& exc) {
+    throw std::runtime_error("Failed to spawn process " + cmd + " exited with an error: " + exc.what());
```

Comment:
i don't guess its possible for there to be stderr here, but I think there might be for line 25 above? so maybe we should try and include it in the exception?